llama.cpp

```diff
@@ -4329,6 +4351,32 @@ static void llm_load_print_meta(llama_model_loader & ml, llama_model & model) {
     if (vocab.linefeed_id    != -1) { LLAMA_LOG_INFO( "%s: LF token         = %d '%s'\n", __func__, vocab.linefeed_id,    vocab.id_to_token[vocab.linefeed_id].text.c_str() );    }
 }
 
+static int llm_determine_max_ngl(const llama_model_loader & ml, const llama_model & model, const int main_gpu) {
+    const auto & hparams = model.hparams;
+
+    size_t available_gpu_memory = llama_get_available_device_memory(main_gpu);
```

Comment:
The program logic here is inconsistent with the `--help`. The `--help` says "based on VRAM size" which I would interpret as total VRAM but here you are using how much *free* VRAM there is.